packages/field-plugin/src/createFieldPlugin/createPluginActions/createPluginActions.ts

```diff
@@ -114,45 +96,42 @@ export const createPluginActions: CreatePluginActions = (
 
   return {
     actions: {
-      setContent: (action) => {
-        const content: unknown =
-          // This is not safe: if the user pass a function to setContent(),
-          //  this code assumes that it is an updater function
-          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-          // @ts-ignore
-          typeof action === 'function' ? action(state.content) : action
-        postToContainer(valueChangeMessage(uid, content))
-        state = {
-          ...state,
-          content,
-        }
-        onUpdateState(state)
+      setContent: (content) => {
+        return new Promise((resolve) => {
+          const callbackId = pushCallback('state', (message) =>
+            resolve(pluginStateFromStateChangeMessage(message)),
+          )
+          postToContainer(
+            valueChangeMessage({ uid, callbackId, model: content }),
```

Comment:
it'd be nice to rename this to contentChangeMessage in the future (not in this PR). It's a note for myself :)

Reply:
Yes, that is true. (@demetriusfeijoo We started with the word "value" and switched to "content")